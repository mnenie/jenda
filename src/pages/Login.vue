<script setup lang="ts">
import { BgPanel, SignInForm } from '@/features/auth';
import { AuthContainer } from '@/widgets/auth';
import { useHead } from '@unhead/vue';

useHead({
  title: 'Jenda | Login'
});
</script>

<template>
  <div :class="$style.login">
    <AuthContainer>
      <SignInForm />
    </AuthContainer>
    <BgPanel />
  </div>
</template>

<style module lang="scss">
.login {
  height: 100%;
  width: 100%;
  display: flex;
}
</style>
