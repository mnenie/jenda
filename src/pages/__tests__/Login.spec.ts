import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import Login from '../Login.vue';
import { useHead } from '@unhead/vue';

vi.mock('@unhead/vue', () => ({
  useHead: vi.fn()
}));

describe('tests for Login.vue', () => {
  const wrapper = shallowMount(Login);

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should has correct head', () => {
    expect(useHead).toHaveBeenCalledWith({
      title: 'Jenda | Login'
    });
  });
});
