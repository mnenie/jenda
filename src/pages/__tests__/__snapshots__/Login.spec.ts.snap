// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for Home.vue > should be render correctly 1`] = `
"<div class="login">
  <auth-container-stub></auth-container-stub>
  <bg-panel-stub></bg-panel-stub>
</div>"
`;

exports[`tests for Login.vue > should be render correctly 1`] = `
"<div class="login">
  <auth-container-stub></auth-container-stub>
  <bg-panel-stub></bg-panel-stub>
</div>"
`;

exports[`tests for Registration.vue > should be render correctly 1`] = `
"<div class="login">
  <auth-container-stub></auth-container-stub>
  <bg-panel-stub></bg-panel-stub>
</div>"
`;
